StackOverflow: Scheduler - _fillNew

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/5dc46eb0-adec-4e38-b5af-c37e7c5d3a87

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:399)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.StackOverflowError: stack size 1039KB
at com.ichi2.utils.JSONArray.arrayToArray(Unknown Source:0)
at com.ichi2.utils.JSONObject.getJSONArray(JSONObject.java:2)
at com.ichi2.libanki.Decks.active(Decks.java:1)
at com.ichi2.libanki.sched.SchedV2._walkingCount(SchedV2.java:2)
at com.ichi2.libanki.sched.Sched._resetRevCount(Sched.java:1)
at com.ichi2.libanki.sched.Sched._resetRev(Sched.java:1)
at com.ichi2.libanki.sched.SchedV2._resetNew(SchedV2.java:1)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:22)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:23)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:23)
[snipped]
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:23)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:23)
at com.ichi2.libanki.sched.SchedV2._fillNew(SchedV2.java:1)
at com.ichi2.libanki.sched.SchedV2._getNewCard(SchedV2.java:1)
at com.ichi2.libanki.sched.Sched._getCard(Sched.java:3)
at com.ichi2.libanki.sched.SchedV2.getCard(SchedV2.java:4)
at com.ichi2.async.CollectionTask.doInBackgroundAnswerCard(CollectionTask.java:10)
at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.java:51)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:2)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
at android.os.AsyncTask$3.call(AsyncTask.java:378)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 4 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
@@ -614,9 +614,9 @@ public class SchedV2 extends AbstractSched {
      * New cards **************************************************************** *******************************
      */
 
-    /** Same as _resetRev, but assume discardCard is currently in the reviewer and so don't conunt it.*/
+    /** Same as _resetNew, but assume discardCard is currently in the reviewer and so don't conunt it.*/
     protected void _resetNew(@Nullable Card discardCard) {
-        _resetRev();
+        _resetNew();
         if (discardCard != null && discardCard.getQueue() == Consts.QUEUE_TYPE_NEW) {
             mNewCount--;
         }

